### DISCUSSION TITLE
Miss `main` field in package.json in npm@3

### DISCUSSION THREAD

**Discussion by nick-ChenZe** [verdaccio/verdaccio#3101] [Q&A]
Hi, i have encountered a problem for the last two days. There is a require error during build process, like
<img width="519" alt="image" src="https://user-images.githubusercontent.com/12672355/160518412-46815825-0600-4ffe-bd39-4abad7c36e16.png">

After locate this problem, i found when i use node@6, npm@3, sometime it will miss `main` field in package.json, so the node application can't not resolve the require statement.

You can use these commands to reproduce this
```
n 6  # or any other node version manager
npm install fis-preprocessor-annotate@0.0.2 --registry=http://0.0.0.0:8080 --verbose
cat node_modules/source-map/node_modules/amdefine/package.json | grep main
```
It work correct in stable node version, so i'm curious about the reason, why the field missed.

Thank you.

**Comment by juanpicado** [comment]
I see https://registry.verdaccio.org/fis-preprocessor-annotate the main field on the body. That might be local issue since verdaccio does not modify the manfiest or remove main field.

Why npm3? anyway does works fine on npm6. I will move this to a discussion because is not a bug.
